semistruct.go

```diff
@@ -0,0 +1,271 @@
+package semistruct
+
+import "strconv"
+import p "github.com/andyleap/parser"
+
+// SemistructLog is the data structure representing the result of a
+// successful semistructured log line parse.
+type SemistructLog struct {
+	Priority int64
+	Tags     []string
+	Attrs    map[string]string
+}
+
+type kvPair struct {
+	key string
+	val string
+}
+
+// NewParseSemistruct instantiates a parser composed of combinators
+// for parsing a semistructured log line into the SemistructLog struct
+// data type.
+func NewParseSemistruct() *p.Grammar {
```

Comment:
sorry for the churn, but maybe NewParser would be better since the package is semistruct.

`semistruct.NewParser()` would be more idiomatic.